File: app/src/main/java/me/jdvp/androidaspectexample/viewmodel/impl/ViewModelModule.java
package me.jdvp.androidaspectexample.viewmodel.impl;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import me.jdvp.androidaspectexample.viewmodel.LoggingViewModel;

/**
 * Module that provides ViewModel dependencies. Also includes lower-level modules. In this case, we are including a module
 * for Services but if we wanted to also do caching, we might have another layer between the VM and services.
 *
 * Created by jdvp on 6/18/18.
 */

@Module
public class ViewModelModule {
    @Provides
    @Singleton
    LoggingViewModel loggingViewModel() {
        return new LoggingViewModelImpl();
    }
}

File: app/src/main/java/me/jdvp/androidaspectexample/viewmodel/impl/ViewModelModule.kt
package me.jdvp.androidaspectexample.viewmodel.impl

import dagger.Module
import dagger.Provides
import me.jdvp.androidaspectexample.viewmodel.LoggingViewModel
import javax.inject.Singleton

/**
 * Module that provides ViewModel dependencies. Also includes lower-level modules (not applicable in this
 * example application as we have no networking or caching).
 */
@Module
class ViewModelModule {
    @Provides
    @Singleton
    fun loggingViewModel(): LoggingViewModel {
        return LoggingViewModelImpl()
    }
}